File: Python/424. Longest Repeating Character Replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        m, r, t = 0, 0, Counter()
        for i in range(len(s)):
            t[s[i]] += 1
            m = max(t.values())
            if r - m < k: r += 1
            else: t[s[i - r]] -= 1
        return r

File: Java/424. Longest Repeating Character Replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int m = 0, r = 0;
        int[] t = new int[26];
        for(int i = 0; i < s.length(); i++){
            t[s.charAt(i) - 'A'] += 1;
            m = Math.max(m, t[s.charAt(i) - 'A']);
            if(r - m < k) r += 1;
            else t[s.charAt(i - r) - 'A'] -= 1;
        }
        return r;
    }
}
